src/main/scala/com.snowplowanalytics.sauna/observers/LocalObserver.scala

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2015 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.sauna.observers
+
+import java.io.{FileInputStream, IOException}
+import java.nio.file._
+import java.nio.file.attribute._
+
+import com.snowplowanalytics.sauna.loggers.Logger
+import com.snowplowanalytics.sauna.processors.Processor
+
+import scala.collection.JavaConversions._
+import scala.collection.mutable
+
+/**
+  * Observes files in local filesystem.
+  */
+class LocalObserver(observedDir: String,
+                    processors: Seq[Processor]) extends Observer { self: Logger =>
+
+  private def processEvent(event: WatchEvent.Kind[Path], path: Path): Unit = {
```

Comment:
Not applicable for this case probably, but it's usually very handy to use `private[package]`, then this member will be available for testing.